save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt

```diff
@@ -33,12 +34,19 @@ abstract class Plugin(
     /**
      * Perform plugin's work.
      *
+     * @param rootDir - test root directory (highest direstory with save.toml)
+     * @param excludedTests - list with excluded tests (relative paths)
      * @return a sequence of [TestResult]s for each group of test resources
      */
-    fun execute(): Sequence<TestResult> {
+    fun execute(rootDir: Path, excludedTests: List<String>?): Sequence<TestResult> {
         clean()
-        // todo: pass individual groups of files to handleFiles? Or it will play bad with batch mode?
+
         val testFilesSequence = discoverTestFiles(testConfig.directory)
+            // removing excluded test resources
+            .filterNot {
+                isExcludedTest(it, rootDir, excludedTests)
```

Comment:
I think it should be moved closer to the place where we filter based on `testFiles` (in `discoverTestFiles`)